WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostDetailFragment.java

```diff
@@ -520,40 +539,84 @@ public void onEventMainThread(ReaderEvents.RelatedPostsUpdated event) {
      * show the passed list of related posts - can be either global (related posts from
      * across wp.com) or local (related posts from the same site as the current post)
      */
-    private void showRelatedPosts(@NonNull ReaderRelatedPostList relatedPosts, boolean isGlobal) {
+    private void showRelatedPosts(@NonNull ReaderRelatedPostList relatedPosts, final boolean isGlobal) {
         // different container views for global/local related posts
         int id = isGlobal ? R.id.related_posts_view_global : R.id.related_posts_view_local;
-        ReaderRelatedPostsView relatedPostsView = (ReaderRelatedPostsView) getView().findViewById(id);
+        ReaderRelatedPostsView relatedPostsView = isGlobal ? mLocalRelatedPostsView : mGlobalRelatedPostsView;
         relatedPostsView.showRelatedPosts(relatedPosts, mPost.getBlogName(), isGlobal);
 
         // tapping a related posts should open the related post detail
         relatedPostsView.setOnRelatedPostClickListener(new ReaderRelatedPostsView.OnRelatedPostClickListener() {
             @Override
             public void onRelatedPostClick(View v, long siteId, long postId) {
-                showRelatedPostDetail(siteId, postId);
+                showRelatedPostDetail(siteId, postId, isGlobal);
             }
         });
 
+        // fade in this related posts view
         if (relatedPostsView.getVisibility() != View.VISIBLE) {
             AniUtils.fadeIn(relatedPostsView, AniUtils.Duration.MEDIUM);
         }
+
+        trackRelatedPostsIfShowing();
+    }
+
+    /*
+     * track that related posts have loaded and are scrolled into view if we haven't
+     * already tracked it
+     */
+    private void trackRelatedPostsIfShowing() {
```

Comment:
Argh, looks like I lost [this line](https://github.com/wordpress-mobile/WordPress-Android/blob/34d584ad1c8ec156e3b602c9375d04f04d3c9faa/WordPress/src/main/java/org/wordpress/android/ui/reader/ReaderPostDetailFragment.java#L1065) when I merged `develop` into this branch. Re-added in 3b4303f